src/router/index.js

```diff
@@ -85,3 +81,16 @@ export default new Router({
     return { x: 0, y: 0 }
   }
 })
+
+router.beforeEach(async(to, from, next) => {
+  const requiresAuth = to.matched.some(rec => rec.meta.requiresAuth)
```

Comment:
Consulting the docs, `const requiresAuth = to.meta.requiresAuth` seems sufficient.
Since Vue Router provides $route.meta that is a *non-recursive merge of all meta fields from parent to child*.
![image](https://user-images.githubusercontent.com/39425103/193565848-11e39572-3c1c-4509-8574-95ba3e953db8.png)
We can change line 86 as such.

Reply:
Reference: https://router.vuejs.org/guide/advanced/meta.html

Reply:
Changed in https://github.com/NCTU-SYNC/sync-frontend/pull/62/commits/743c71f6499a0a06155167aed76085d488d78650